docs: single project example (#29)
infracost/infracost-atlantis: examples/single_project/README.md

This repo.yaml file contains a workflow specification to use Infracost with a single project. It uses `infracost comment` to post Infracost results to your PR. 

## Running with GitHub

> 1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) dockerhub image

Reviewer: ```suggestion
1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) Docker image
```

2. You'll need to pass the following custom env vars into the container
   ```sh
   GITHUB_TOKEN=<your-github-token>
   INFRACOST_API_KEY=<your-infracost-api-token>
   ```
3. Add the following yaml spec to `repos.yaml` or `atlantis.yaml` config files:
   ```yaml
   repos:
     - id: /.*/
       workflow: terraform-infracost
       pre_workflow_hooks:
         # Clean up any files left over from the last run

>          - run: rm /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json

Reviewer 2: Are we sure all these envs are defined at the time of pre-workflow-hook run?  Seems like WORKSPACE & REPO_REL_DIR might be empty so maybe this should be moved to the plan: steps:?  If it's needed at all?

   workflows:
     terraform-infracost:
       plan:
         steps:
           - init
           - plan
           # Run Infracost breakdown and save to a tempfile, namespaced by this project, PR, workspace and dir
           - run: infracost breakdown --path=$PLANFILE --format=json --log-level=info --out-file=/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json
           # Use Infracost comment to create a comment containing the results for this project.
           - run: infracost comment github --repo $BASE_REPO_OWNER/$BASE_REPO_NAME --pull-request $PULL_NUM --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json --github-token $GITHUB_TOKEN
   ```

> 4. Restart the atlantis application with the new env vars and config 
> 5. Send a pull request in GitHub to change something in TF, the Infracost pull request comment should be added.

Reviewer: ```suggestion
4. Restart the Atlantis application with the new env vars and config 
5. Send a pull request in GitHub to change something in the Terraform code, the Infracost pull request comment should be added.
```

> ## Running with Gitlab

Reviewer: ```suggestion
## Running with GitLab
```

> 1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) dockerhub image

Reviewer: ```suggestion
1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) Docker image
```

2. You'll need to pass the following custom env vars into the container
   ```sh
   GITLAB_TOKEN=<your-gitlab-token>
   INFRACOST_API_KEY=<your-infracost-api-token>
   ```
3. Add the following yaml spec to `repos.yaml` or `atlantis.yaml` config files:
   ```yaml
   repos:
     - id: /.*/
       workflow: terraform-infracost
       pre_workflow_hooks:
         # Clean up any files left over from the last run

>          - run: rm /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json

Reviewer: Should we do this at the end of the workflow now instead of here? I think we only needed it here for the multi-project example before since we didn't have a post workflow hook.

Author: Yeah let's move to the end of the run

   workflows:
     terraform-infracost:
       plan:
         steps:
           - init
           - plan
           # Run Infracost breakdown and save to a tempfile, namespaced by this project, PR, workspace and dir
           - run: infracost breakdown --path=$PLANFILE --format=json --log-level=info --out-file=/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json
           # Use Infracost comment to create a comment containing the results for this project.
           - run: infracost comment gitlab --repo $BASE_REPO_OWNER/$BASE_REPO_NAME --merge-request $PULL_NUM --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json --gitlab-token $GITLAB_TOKEN
   ```  

> 4. Restart the atlantis application with the new env vars and config
> 5. Send a merge request in GitLab to change something in TF, the Infracost merge request comment should be added.

Reviewer: ```suggestion
4. Restart the Atlantis application with the new env vars and config
5. Send a merge request in GitLab to change something in the Terraform code, the Infracost merge request comment should be added.
```
